frontend/model/contracts/group.js

```diff
@@ -261,38 +261,79 @@ function updateGroupStreaks ({ state, getters }) {
   }
 }
 
+const removeGroupChatroomProfile = (state, chatRoomID, member) => {
+  console.log('jjjj removeGroupChatroomProfile', JSON.parse(JSON.stringify({ state, chatRoomID, member })))
+  Vue.set(state.chatRooms[chatRoomID], 'users',
+    Object.fromEntries(
+      Object.entries(state.chatRooms[chatRoomID].users)
+        .map(([memberKey, profile]) => {
+          console.log('jjjj removeGroupChatroomProfile ii', {
+            memberKey, member, profile, active: (profile: any)?.status === PROFILE_STATUS.ACTIVE
+          })
+          if (memberKey === member && (profile: any)?.status === PROFILE_STATUS.ACTIVE) {
+            return [memberKey, { status: PROFILE_STATUS.REMOVED }]
+          }
+          return [memberKey, profile]
+        })
+    )
+  )
+}
+
 const leaveChatRoomAction = (state, { chatRoomID, member, leavingGroup }, meta) => {
   const sendingData = leavingGroup
     ? { member: member }
     : { member: member, username: meta.username }
 
+  if (state?.chatRooms?.[chatRoomID]?.users?.[member]?.status !== PROFILE_STATUS.REMOVED) {
+    return
+  }
+
+  const extraParams = {}
+
+  // When a group is being left, we want to also leave chatrooms,
+  // including private chatrooms. Since the user issuing the action
+  // may not be a member of the chatroom, we use the group's CSK
+  // unconditionally in this situation, which should be a key in the
+  // chatroom (either the CSK or the groupKey)
+  if (leavingGroup) {
+    const signingKeyId = sbp('chelonia/contract/currentKeyIdByName', state, 'csk', true)
+
+    // If we don't have a CSK, it is because we've already been removed.
+    // Proceeding would cause an error
+    if (!signingKeyId) {
+      return
+    }
+
+    extraParams.signingKeyId = signingKeyId
+    extraParams.innerSigningContractID = null
+  }
+
   return sbp('gi.actions/chatroom/leave', {
     contractID: chatRoomID,
     data: sendingData,
-    // When a group is being left, we want to also leave chatrooms,
-    // including private chatrooms. Since the user issuing the action
-    // may not be a member of the chatroom, we use the group's CSK
-    // unconditionally in this situation, which should be a key in the
-    // chatroom (either the CSK or the groupKey)
-    ...(leavingGroup && {
-      signingKeyId: sbp('chelonia/contract/currentKeyIdByName', state, 'csk'),
-      innerSigningContractID: null
-    }),
+    ...extraParams,
     hooks: {
       preSendCheck: (_, state) => {
         // Avoid sending a duplicate action if the person has already
         // left or been removed from the chatroom
         return !!state?.users?.[member]
       }
     }
+  }).catch((e) => {
+    console.log('jjj action', { e })
+    if (leavingGroup && e?.name === 'GIErrorUIRuntimeError' && e?.cause?.name === 'GIErrorMissingSigningKeyError') {
```

Comment:
This `.cause` thing doesn't seem to be a thing? `GIErrorMissingSigningKeyError` sets `.name` to `GIErrorMissingSigningKeyError`.